scipy/stats/_continuous_distns.py

```diff
@@ -3582,56 +3583,943 @@ class levy_stable_gen(rv_continuous):
+                else:
+                    p1 = 1 / 2. + beta * K(alpha) / (2. * alpha)
+                return dict(alpha=alpha, p1=p1, scale=scale, loc=loc)
+            else:
+                raise ValueError("Parameters out of the range.")
+
+        def _c_to_b(alpha, delta, scale, loc):
+            if cls._argcheck(alpha, delta, "C"):
+                if alpha==1:
+                    scale = scale / np.sqrt(np.pi**2 / 4. + loc**2)
+                    loc = np.pi * np.tan(delta * np.pi / 2.) / 2.
+                else:
+                    beta = alpha * delta / K(alpha)
+                return dict(alpha=alpha, beta=beta, scale=scale, loc=loc)
+            else:
+                raise ValueError("Parameters out of the range.")
+
+        def _p_to_b(alpha, p1, scale, loc):
+            if cls._argcheck(alpha, p1, "P"):
+                if alpha==1:
+                    scale = scale / np.sqrt(np.pi**2 / 4. + loc**2)
+                    loc = np.pi * np.tan((2 * p1 - 1) * np.pi / 2.) / 2.
+                else:
+                    beta = (2 * p1 - 1) * alpha / K(alpha)
+                return dict(alpha=alpha, beta=beta, scale=scale, loc=loc)
+            else:
+                raise ValueError("Parameters out of the range.")
 
-        def otherwise(alpha, beta, TH, aTH, bTH, cosTH, tanTH, W):
-            # alpha is not 1 and beta is not 0
-            val0 = beta*np.tan(np.pi*alpha/2)
-            th0 = np.arctan(val0)/alpha
-            val3 = W/(cosTH/np.tan(alpha*(th0 + TH)) + np.sin(TH))
-            res3 = val3*((np.cos(aTH) + np.sin(aTH)*tanTH -
-                          val0*(np.sin(aTH) - np.cos(aTH)*tanTH))/W)**(1.0/alpha)
-            return res3
+        # original parameters are first converted to parameterization B
+        # (the best choice due to transformation equations)
+        in_dict = dict(A = _a_to_b,
+                       B = lambda **args: args, # we are passing the params to the function
+                       C = _c_to_b,
+                       P = _p_to_b)
 
-        def alphanot1func(alpha, beta, TH, aTH, bTH, cosTH, tanTH, W):
-            res = _lazywhere(beta == 0,
-                             (alpha, beta, TH, aTH, bTH, cosTH, tanTH, W),
-                             beta0func, f2=otherwise)
-            return res
+        # parameters are converted from parameterization B into the final parameterization
+        out_dict = dict(A = _b_to_a,
+                        B = lambda **args: args, # we are returning the params under new param!
+                        C = _b_to_c,
+                        P = _b_to_p)
 
+        return out_dict[target_param_type](**in_dict[original_param_type](**kwargs))
+
+
+    def _rvs(self, alpha, asymmetry, param='A'):
+        """
+        Generates a vector of random stable variables under parameterization A
+        using Weron's formulation [WE]. It is identical with the implementation
+        in GSL library.
```

Comment:
This docstring suggests the following code might have been [copied from GSL](https://www.gnu.org/software/gsl/). It is described as identical to GSL's implementation.

Reply:
Hi @bsdz, the GSL library's random number generation algorithm is described as identical to Weren's forumation in the libstable paper (Royuela-del-Val et al., 2017). See the bottom of page 8: https://www.jstatsoft.org/article/view/v078i01/v78i01.pdf . We don't need to read any code to identify the method used by a particular library. Even a look at their documentation would suffice.

To allege we are copying code or lying about what we have read is an unnecessary provocation. :)

Reply:
We invite you to collaborate with us as we have before. We give you all the credit for your initiative and prior work. But please do recognize that we have a lot more experience working with the mathematics of stable law distributions, and can identify when methods are wrong. A library like SciPy cannot afford to repeat and propagate mistakes in the community.

Reply:
So GSL documentation claims to be based on the same paper? Then probably this note should be phrased as such.